### PR TITLE
issue #32: added every useful workflows

### LINKED ISSUE #32: Fix every workflow inside this repository
This issue is part of https://github.com/ai-cfia/github-workflows/issues/71

### PR DESCRIPTION
**Pull request by ThomasCardin**
- Closes #32 

Workflow included:
- [X] Lint test
- [X] Markdown validation
- [X] ai-cfia repo standard
- [X] build and push to github container registry (GCR)

### CODE REVIEW

**Comment by ThomasCardin** [comment]
Hey @k-allagbe, I see that this repository doesn't have a description. Could you add one please? Also, for the test, do you have any idea what is the error?

**Comment by k-allagbe** [comment]
> Hey @k-allagbe, I see that this repository doesn't have a description. Could you add one please? Also, for the test, do you have any idea what is the error?

Done.

**Comment by ThomasCardin** [comment]
@k-allagbe  Hey, I wanted to inform you that I've pushed a .vscode folder. It includes extensions and settings to ensure that GitHub actions pass (with a maximum of 80 characters per line for .md files). Additionally, if you have any extensions that you use for this project and keep locally, feel free to add them to the files .vscode/extensions.json or .vscode/settings.json. This will enable the next developer joining the project to have everything they need to start development as quickly as possible.

**Review by k-allagbe — APPROVED** [review]
Looks good to me. Note that I added the extensions to the devcontainer config.

**Comment by k-allagbe** [comment]
@ThomasCardin not sure why this https://github.com/ai-cfia/finesse-backend/actions/runs/7631368304/job/20789157763?pr=33. There is a `TESTING.md` file.

**Comment by ThomasCardin** [comment]
> @ThomasCardin not sure why this https://github.com/ai-cfia/finesse-backend/actions/runs/7631368304/job/20789157763?pr=33. There is a `TESTING.md` file.

Yes, I believe the workflow is checking out the main branch and not this one. In reality, it's not a problem that it checks out the main branch because there should be a TESTING.md by default when creating a directory in the future.

Anyway, you can ignore the error for this PR.

**Comment by rngadam** [comment]
> Yes, I believe the workflow is checking out the main branch and not this one. In reality, it's not a problem that it checks out the main branch because there should be a TESTING.md by default when creating a directory in the future.

@ThomasCardin 
as discussed, Github Workflows check should be done on the current branch to give an opportunity to the developer to fix it and only fix for repos that are under active development.

**Comment by ThomasCardin** [comment]
Closing this PR. We will preview deployment when this issue (https://github.com/ai-cfia/infra/issues/20) will be solved (iteration 3). No more maintenance time will be put into Vercel

**Comment by ThomasCardin** [comment]
@rngadam ping (workflows are fixed)

File: .github/workflows/github-workflows.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Docker build and push to ghcr.io/ai-cfia and lint test
+name: Workflows
```
Comment (rngadam) [inline .github/workflows/github-workflows.yml:1]: python-backend-workflows? it would make it more readable in the Github Actions log interface perhaps?

**Review by rngadam — APPROVED** [review]
please consider one last suggestion made for naming

File: .github/workflows/github-workflows.yml (hunk 1 above)
Comment (ThomasCardin) [reply .github/workflows/github-workflows.yml:1]: Alright, il'l do the same with all frontend workflow. Maybe something like `node-frontend-workflows`

Comment (rngadam) [reply .github/workflows/github-workflows.yml]: react-frontend-workflows would be more accurate IMO

**Pull request merged**